### PR TITLE
V2:Astro-958 - Timeline KM Style Update

### PR DESCRIPTION
**Pull request by im-cr**
**Tickets**
  - https://rocketcom.atlassian.net/browse/ASTRO-958

**Changes** - commit: 
  - theme__light
  - theme__dark
  - rux-timeline
  - rux-timeline-track
  - rux-time-region

**Designs**
  - [Dark](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/046dcb8e-fcdc-42ae-ad3d-205dc019f2fa/files/1d17329d-e1a3-490a-a6ad-097e2ecd4825?filter=changed&s=timeline)
  - [Light](https://app.abstract.com/projects/2d8cf270-0b93-11e8-89ff-61554354acb1/branches/master/files/b7c2bf0c-9dec-4b07-96d4-ead85393c9d0?s=timeline)

### CODE REVIEW

File: src/components/rux-timeline/rux-timeline-region.js [hunk 1]
```diff
@@ -86,33 +86,29 @@ export class RuxTimelineRegion extends LitElement {
...
         height: 60px;
 
         z-index: 1;
-        /* background-color: red; */
         filter: none;
       }
 
       :host([hidden]) {
-        /* filter: saturate(50%); */
         background-color: #203246;
```
Comment (amalyah) [inline src/components/rux-timeline/rux-timeline-region.js:108]: No variable?

File: src/components/rux-timeline/rux-timeline-region.js [hunk 2]
```diff
@@ -86,33 +86,29 @@ export class RuxTimelineRegion extends LitElement {
...
         opacity: 0.8;
         z-index: 0;
         border-color: #203246;
```
Comment (amalyah) [inline src/components/rux-timeline/rux-timeline-region.js:111]: No variable?

File: src/components/rux-timeline/rux-timeline-region.js [hunk 3]
```diff
@@ -86,33 +86,29 @@ export class RuxTimelineRegion extends LitElement {
 
         font-size: 0.875rem;
```
Comment (amalyah) [inline src/components/rux-timeline/rux-timeline-region.js:87]: Change to 1 rem

File: src/components/rux-timeline/rux-timeline-region.js [hunk 4]
```diff
@@ -128,13 +124,12 @@ export class RuxTimelineRegion extends LitElement {
       :host(.current),
       :host([temporality='present']) {
         border: 1px solid #4dacff;
```
Comment (amalyah) [inline src/components/rux-timeline/rux-timeline-region.js:126]: No variable?

File: src/components/rux-timeline/rux-timeline.js [hunk 5]
```diff
@@ -249,7 +249,7 @@ export class RuxTimeline extends LitElement {
 
       .rux-timeline__track__label-container {
         width: 100%;
-        background-color: var(--timelineTrackLabelBackgroundColor, rgb(40, 63, 88));
+        background-color: var(--timelineTrackLabelBackgroundColor);
         font-size: 0.875rem;
```
Comment (amalyah) [inline src/components/rux-timeline/rux-timeline.js:253]: 1 rem

File: src/css/common/_theme--light.css [hunk 6]
```diff
@@ -383,25 +383,22 @@
...
-  );
+  --timelineTrackLabelBackgroundColor: var(--surfaceElements);
+  --timelineTrackBackgroundColor: var(--surfaceElements);
 
-  --timelineRegionBackgroundColor: var(--primaryLighten4);
-  --timelineRegionTextColor: var(--fontColor);
-  --timelineRegionBorderColor: var(--primaryLighten3);
+  --timelineRegionBackgroundColor: var(--backgroundColor);
+  --timelineRegionTextColor: var(--defaultText);
+  --timelineRegionBorderColor: var(--primary);
```
Comment (amalyah) [inline src/css/common/_theme--light.css:397]: Border color actually depends on the status of the event. Only the normal green status gets the primary border. Other statuses get borders based on their status colors (example: Critical red colored border with Critical status).

True for both dark and light themes.

**Review by amalyah — CHANGES_REQUESTED** [review]

File: src/components/rux-timeline/rux-timeline-region.js (hunk 3 above)
Comment (im-cr) [reply src/components/rux-timeline/rux-timeline-region.js:87]: resolved

File: src/components/rux-timeline/rux-timeline.js (hunk 5 above)
Comment (im-cr) [reply src/components/rux-timeline/rux-timeline.js]: resolved

**Comment by im-cr** [comment]
@amalyah That #203246 color isn't represented in either dark or light css themes. I left as is for coverage in both versions because there is no timeline specific css file and because it's on a hidden element.

File: src/components/rux-timeline/rux-timeline-region.js (hunk 2 above)
Comment (im-cr) [reply src/components/rux-timeline/rux-timeline-region.js:111]: See General Comment https://github.com/RocketCommunicationsInc/astro-components/pull/58#issuecomment-745478572

File: src/components/rux-timeline/rux-timeline-region.js (hunk 4 above)
Comment (im-cr) [reply src/components/rux-timeline/rux-timeline-region.js]: General comment https://github.com/RocketCommunicationsInc/astro-components/pull/58#issuecomment-745478572

File: src/css/common/_theme--light.css (hunk 6 above)
Comment (im-cr) [reply src/css/common/_theme--light.css:394]: @amalyah got it. Updated and added in. Border color status (normal/critical/serious/caution/standby/off) has been added. 

For confirmation, border color on selected region elements should stay consistent with selected design state and not conform to unselected status border color correct? This is how I've currently implemented.

**Comment by amalyah** [comment]
Sorry, looks like I messed up which pieces are supposed to be 16 px vs. not.
- Track Label: 16px 
- Region Label: 14px
- Region Time: 12px
- Timeline Ruler: 16px

File: src/css/common/_theme--light.css (hunk 6 above)
Comment (amalyah) [reply src/css/common/_theme--light.css:394]: Correct

**Review by amalyah — APPROVED** [review]

**Pull request merged**